src/ode_solver/pod_galerkin_ode_solver.cpp

```diff
@@ -36,39 +120,105 @@ void PODGalerkinODESolver<dim,real,MeshType>::step_in_time (real dt, const bool
     //Galerkin projection, pod_basis = V
     //V^T*J*V*p = -V^T*R
 
-    pod->getPODBasis()->Tvmult(*this->reduced_rhs, this->dg->right_hand_side); // reduced_rhs = (pod_basis)^T * right_hand_side
-
-    pod->getPODBasis()->Tmmult(*this->reduced_lhs_tmp, this->dg->system_matrix); //reduced_lhs_tmp = pod_basis^T * system_matrix
-
-    this->reduced_lhs_tmp->mmult(*this->reduced_lhs, *pod->getPODBasis()); // reduced_lhs = reduced_lhs_tmp*pod_basis
-
-    solve_linear(
-            *this->reduced_lhs,
-            *this->reduced_rhs,
-            *this->reduced_solution_update,
-            this->ODESolverBase<dim,real,MeshType>::all_parameters->linear_solver_param);
-
-    pod->getPODBasis()->vmult(this->solution_update, *this->reduced_solution_update);
+    const Epetra_CrsMatrix epetra_system_matrix = this->dg->system_matrix.trilinos_matrix();
```

Comment:
`Epetra_CrsMatrix` has a virtual destructor too. Not sure if it would result in the same issue that led to switching to pointers for partial second derivatives of functional.

Reply:
It should be fine. The issue with the second partial derivatives of Functional seems to happen because they are member variables.